proxy/freedom/freedom.go

```diff
@@ -321,12 +327,24 @@ func NewPacketWriter(conn net.Conn, h *Handler, ctx context.Context, UDPOverride
 		counter = statConn.WriteCounter
 	}
 	if c, ok := iConn.(*internet.PacketConnWrapper); ok {
-		return &PacketWriter{
-			PacketConnWrapper: c,
-			Counter:           counter,
-			Handler:           h,
-			Context:           ctx,
-			UDPOverride:       UDPOverride,
+		if hasNoise {
+			return &NoisePacketWriter{
```

Comment:
i suggest to structure the code similarly to FragmentWriter, and add this wrapper outside of this function

Reply:
I am almost done , I clean up the code a lot,for this one I basicly had to create another function to return the wrapper,this PacketWriter uses counter,I do not know what it does(when i removed it it worked fine but it might mess something up so I kept it),for fragment they basicly added the wrapper in prooccess function,But in my case if I do that the code will get messy so I  added another function(copy NewPacketWriter and return my wrapper instead)